source/common/network/connection_impl.cc

```diff
@@ -279,13 +279,8 @@ void ConnectionImpl::enableHalfClose(bool enabled) {
 }
 
 void ConnectionImpl::readDisable(bool disable) {
-  ASSERT(state() == State::Open);
```

Comment:
High level comment - prior to this the thought was that changing the read status of an open socket indicated an error and should be avoided.  I'm fine having error handling in case someone makes an error, but do we also think we should leave this ASSERT in and fix this at the call site to not attempt to change readDisable status for closed sockets?  It seems buggy behavior to read disable or enable a closed socket as it won't actually work.

Reply:
We may want to go further and say it is illegal to call readEnabled() on a closed socket.  We'll need to add some connection state checks in a few places to avoid hitting ASSERT failures during runtime.

Reply:
I brought the ASSERT back and extended it to also cover file_event_ != nullptr.  ASSERT also added to readEnabled.

connection_.state() check added to source/common/http/http1/codec_impl.cc to avoid hitting the ASSERT in ConnectionImplTest.ReadDisableAfterCloseHandledGracefully